File: app/src/main/java/com/example/loveproposeapps/MainActivity2.java
package com.example.loveproposeapps;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity2 extends AppCompatActivity {

    private EditText male,female;
    TextView tv;
    String result;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
    }



    public void showResult(View view) {


        male = findViewById(R.id.editText1ID);
        female = findViewById(R.id.editText2ID);
        tv= findViewById(R.id.calculateetextI);


        String f = female.getText().toString();
        String m = male.getText().toString();
        String sum = m+f;
        sum = sum.toLowerCase();

        int value = sum.hashCode();
        Random random = new Random(value);
        result = (random.nextInt(100)+1)+"%";

        if (f.equals("") || m.equals("")){

            Toast.makeText(MainActivity2.this,"Enter Names",Toast.LENGTH_SHORT).show();
        }
        tv.setText(result);
    }
}

File: app/src/main/java/com/example/myapplication/MainActivity2.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.widget.EditText
import android.widget.TextView
import android.os.Bundle
import android.view.View
import com.example.myapplication.R
import android.widget.Toast
import java.util.*

class MainActivity2 : AppCompatActivity() {
    private var male: EditText? = null
    private var female: EditText? = null
    var tv: TextView? = null
    var result: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
    }

    fun showResult(view: View?) {
        male = findViewById(R.id.editText1ID)
        female = findViewById(R.id.editText2ID)
        tv = findViewById(R.id.calculateetextI)
        val f = female?.getText().toString()
        val m = male?.getText().toString()
        var sum = m + f
        sum = sum.lowercase(Locale.getDefault())
        val value = sum.hashCode()
        val random = Random(value.toLong())
        result = (random.nextInt(100) + 1).toString() + "%"
        if (f == "" || m == "") {
            Toast.makeText(this@MainActivity2, "Enter Names", Toast.LENGTH_SHORT).show()
        }
        tv?.setText(result)
    }
}